engine/src/main/java/org/pentaho/di/job/entries/copymoveresultfilenames/JobEntryCopyMoveResultFilenames.java

```diff
@@ -0,0 +1,702 @@
+      add_date = rep.getJobEntryAttributeBoolean( id_jobentry, "add_date" );
+      add_time = rep.getJobEntryAttributeBoolean( id_jobentry, "add_time" );
+      SpecifyFormat = rep.getJobEntryAttributeBoolean( id_jobentry, "SpecifyFormat" );
+      date_time_format = rep.getJobEntryAttributeString( id_jobentry, "date_time_format" );
+      AddDateBeforeExtension = rep.getJobEntryAttributeBoolean( id_jobentry, "AddDateBeforeExtension" );
+      action = rep.getJobEntryAttributeString( id_jobentry, "action" );
+
+      OverwriteFile = rep.getJobEntryAttributeBoolean( id_jobentry, "OverwriteFile" );
+      CreateDestinationFolder = rep.getJobEntryAttributeBoolean( id_jobentry, "CreateDestinationFolder" );
+      RemovedSourceFilename = rep.getJobEntryAttributeBoolean( id_jobentry, "RemovedSourceFilename" );
+      AddDestinationFilename = rep.getJobEntryAttributeBoolean( id_jobentry, "AddDestinationFilename" );
+
+    } catch ( KettleException dbe ) {
+      throw new KettleXMLException( BaseMessages.getString(
+        PKG, "JobEntryCopyMoveResultFilenames.CanNotLoadFromRep", "" + id_jobentry, dbe.getMessage() ) );
+    }
+  }
+
+  public void saveRep( Repository rep, IMetaStore metaStore, ObjectId id_job ) throws KettleException {
+    try {
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "foldername", foldername );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "specify_wildcard", specifywildcard );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "wildcard", wildcard );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "wildcardexclude", wildcardexclude );
+
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "destination_folder", destination_folder );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "nr_errors_less_than", nr_errors_less_than );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "success_condition", success_condition );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "add_date", add_date );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "add_time", add_time );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "SpecifyFormat", SpecifyFormat );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "date_time_format", date_time_format );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "AddDateBeforeExtension", AddDateBeforeExtension );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "action", action );
+
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "OverwriteFile", OverwriteFile );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "CreateDestinationFolder", CreateDestinationFolder );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "RemovedSourceFilename", RemovedSourceFilename );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "AddDestinationFilename", AddDestinationFilename );
+
+    } catch ( KettleDatabaseException dbe ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)